build/build.go

```diff
@@ -95,9 +95,12 @@ func parseIncompleteSecret(secret *v1.Secret) *IncompleteSecret {
 	warningMsg := "Warning: Incomplete secret, please replace it with the secret content"
 	if secret.Type == v1.SecretTypeTLS {
```

Comment:
can you add to the yaml encode test case the SecretTypeBasicAuth and IncompleteSecret? Actually only the generic case is covered.